Cookbook/Technical-Documents/CRP-Bot-ImplementationDetails.md

```diff
@@ -0,0 +1,108 @@
+* Then call `JiraService.document(from: [ChangeLogSections]` to build a [JIRA document](https://developer.atlassian.com/cloud/jira/platform/apis/document/structure/) (a JIRA representation of formatted text/documents containing headings, bullet points, links, etc)
+
+This method uses helpers like `formatMessageLine` to transform a commit message into a `FieldType.TextArea.ListItem` – aka a model objet representing a bullet point item in the JIRA formatted text.
+
+Those `FieldType.TextArea` subtypes, used to represent various entities in a formatted text (textarea) in JIRA, are all declared in the bot's framework code in `JiraService+FieldTypes.swift`. Conforming to `Content` (which itself inherits from `Codable`), they can then be transformed into JSON when included in the API call to create the CRP ticket.
+
+### `createAndSetFixedVersions`
+
+This method is declared in `JiraService+CRP.swift` and its goal is to iterate over each  `ChangelogSection` (representing a given board with its associated tickets included in the CRP), and for each board/section that is whitelisted in our static `self.knownProjects` list:
+
+* Create an apptly named JIRA release if a release with that name doesn't already exist; otherwise, reuse the existing one
+* Then for each ticket in this `ChangelogSection` (i.e. associated with this board), make an API request to set the `Fix Version` field to this JIRA Release. This is done in `batchSetFixedVersions(…)` and described in the following section
+
+Any section that is not part of the `knownProjects` list (static constant declared in `configure.swift`) will be skipped during this process. Even though tickets on unknown boards will be included in the CRP ticket's body, there will be no attempt to create a JIRA version on any unknown board nor touch the Fix Version field of those tickets. 
+
+This whitelist has two main purposes:
+
+1. First, to avoid accidentally creating release and touch tickets that are not supposed to be our responsibility and might have been wrongfully detected when parsing the commit messages. For example if a commit message contains a reference to the original ticket but also mentions another ticket from a board we don't own, we don't want to process the latter one and affect a board and ticket we don't own
+2. Some boards, even if they are supposed to follow the CRP process, are misconfigured in JIRA and use a custom set of fields and custom process instead of using the same fields and configuration than all the other boards. 
+
+Those misconfigured boards are tracked in [IOSP-101](https://babylonpartners.atlassian.net/browse/IOSP-101). 
+
+Trying to execute the CRP automation on those boards would end in an invalid API request or worse, some unexpected changes in the tickets, because field IDs would not match the ones expected by the bot.
```

Comment:
```suggestion
Trying to execute the CRP automation on those boards would end in an invalid API request or worse, some unexpected changes in the tickets, because field IDs would not match the ones expected by the bot. Which is why those boards are not included in the whitelist.
```